src/smt/theory_str_noodler/formula.h

```diff
@@ -523,6 +523,15 @@ namespace smt::noodler {
             this->predicates = new_predicates;
         } 
 
+        bool contains_type(PredicateType type) const {
```

Comment:
`is_contains_type` sounds more like: is there a contains predicate? I wanted to say: Does the Formula contain a predicate of a certain type?

Reply:
`contains_pred_type()`? `contains_pred_of_type()`? I would be for more explicit names as often as possible.

Additionally, since we seem to have problems with this function, a comment with `Does the Formula contain a predicate of a certain type?` would be helpful.

Reply:
Ah ok, I would add a comment to make it more obvious.

Reply:
Added a comment